packages/integrations/useIDBKeyval/index.ts

```diff
@@ -0,0 +1,87 @@
+import type { ConfigurableFlush, MaybeComputedRef, RemovableRef } from '@vueuse/shared'
+import { resolveUnref } from '@vueuse/shared'
+import type { Ref } from 'vue-demi'
+import { ref, shallowRef, watch } from 'vue-demi'
+import { del, get, set, update } from 'idb-keyval'
+
+export interface UseIDBOptions extends ConfigurableFlush {
+  /**
+   * Watch for deep changes
+   *
+   * @default true
+   */
+  deep?: boolean
+
+  /**
+   * On error callback
+   *
+   * Default log error to `console.error`
+   */
+  onError?: (error: unknown) => void
+
+  /**
+   * Use shallow ref as reference
+   *
+   * @default false
+   */
+  shallow?: boolean
+}
+
+/**
+ *
+ * @param key
+ * @param initialValue
+ * @param options
+ */
+export function useIDBKeyval<T>(
+  key: IDBValidKey,
+  initialValue: MaybeComputedRef<T>,
+  options: UseIDBOptions = {},
+): RemovableRef<T> {
+  const {
+    flush = 'pre',
+    deep = true,
+    shallow,
+    onError = (e) => {
+      console.error(e)
+    },
+  } = options
+
+  const data = (shallow ? shallowRef : ref)(initialValue) as Ref<T>
+
+  const rawInit: T = resolveUnref(initialValue)
+
+  async function read() {
+    try {
+      const rawValue = await get<T>(key)
+      if (rawValue === undefined) {
+        if (rawInit)
+          set(key, rawInit)
+      }
+      else {
+        data.value = rawValue
+      }
+    }
+    catch (e) {
+      onError(e)
+    }
+  }
+
+  read()
+
+  async function write() {
+    try {
+      if (data.value == null)
+        await del(key)
+      else
+        await update(key, () => ({ ...data.value }))
```

Comment:
Question, did this have the same issue that you were messaging me about? The issue with `...someString` etc?

Reply:
yeah looks like that one is having an issue too. This is what jimmy and I added when we worked through the bug the other day, it had something to do with proxy objects not able to be stored in index db if I remember correctly.

Reply:
changed this logic to only make copy with spread syntax if data.value is an object or array